src/compile/selection/transforms/translate.ts

```diff
@@ -32,14 +31,12 @@ const translate:TransformCompiler = {
       value: {},
       on: [{
         events: events.map((e) => e.between[0]),
-        update: '{x: x(unit), y: y(unit), ' +
-          `width: ${size}.width, height: ${size}.height, ` +
+        update: '{x: x(unit), y: y(unit)' +
+          (x !== null ? ', extent_x: ' + (hasScales ? domain(model, X) :
```

Comment:
We no longer need to determine them from the `unit` signal because I now call `model.getSizeSignalRef`.